pkg/plugin/keyutil/pem_test.go

```diff
@@ -49,12 +52,12 @@ func TestReadRSAPrivateKey(t *testing.T) {
 			t.Parallel()
 
 			gotPK, err := ReadRSAPrivateKey(tc.pkPEMString)
-			if err != nil {
-				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
-					t.Errorf(diff)
-				}
-			} else {
-				if diff := cmp.Diff(gotPK, testPrivateRSAKey); diff != "" {
+
+			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+				t.Errorf(diff)
+			}
+			if err == nil {
```

Comment:
I still don't think we need this check. If there's an error, cmp will be diffing two `nil`s which will be true.

Reply:
Yes, but the pk is type `*rsa.PrivateKey`, `cmp.Diff` on that when it's nil will casue panic.

Reply:
Can you share the panic?

Reply:
if the code is like this:
```
if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
    t.Errorf(diff)
}
if diff := cmp.Diff(gotPK, tc.wantPK); diff != "" {
    t.Errorf("rsa private key got unexpected diff (-want,+got):\n%s", diff)
}
```
The panic for test case `invalid_pem` will be like:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5e6b32]

goroutine 23 [running]:
testing.tRunner.func1.2({0x8f27a0, 0xdb7c90})
        /usr/local/google/home/qinhang/Development/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
        /usr/local/google/home/qinhang/Development/go/src/testing/testing.go:1548 +0x397
panic({0x8f27a0?, 0xdb7c90?})
        /usr/local/google/home/qinhang/Development/go/src/runtime/panic.go:914 +0x21f
crypto/rsa.(*PrivateKey).Equal(0x0, {0x943240?, 0x0})
        /usr/local/google/home/qinhang/Development/go/src/crypto/rsa/rsa.go:132 +0x32
reflect.Value.call({0xc0000b2050?, 0xc00007c078?, 0x7fee257853d0?}, {0x98c54b, 0x4}, {0xc000185b20, 0x2, 0x447e8b?})
        /usr/local/google/home/qinhang/Development/go/src/reflect/value.go:596 +0xce7
reflect.Value.Call({0xc0000b2050?, 0xc00007c078?, 0x0?}, {0xc000185b20?, 0xa609c8?, 0xc0000b2050?})
        /usr/local/google/home/qinhang/Development/go/src/reflect/value.go:380 +0xb9
github.com/google/go-cmp/cmp.(*state).callTTBFunc(0x0?, {0xc0000b2050?, 0xc00007c078?, 0xc0000b2050?}, {0x943240?, 0x0?, 0x1?}, {0x943240, 0x0, 0x16})
        /usr/local/google/home/qinhang/Development/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:359 +0x28d
github.com/google/go-cmp/cmp.(*state).tryMethod(0xc0000ac000?, {0xa609c8?, 0x943240?}, {0x943240?, 0x0?, 0x7fee25787fc8?}, {0x943240?, 0x0?, 0x0?})
        /usr/local/google/home/qinhang/Development/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:319 +0x17d
github.com/google/go-cmp/cmp.(*state).compareAny(0xc0000ac000, {0xa57ae0, 0xc0000b0000?})
        /usr/local/google/home/qinhang/Development/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:266 +0x539
github.com/google/go-cmp/cmp.Diff({0x943240, 0x0}, {0x943240, 0x0}, {0x0?, 0x525f3c?, 0x52658e?})
        /usr/local/google/home/qinhang/Development/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:122 +0x75
github.com/abcxyz/jvs-plugin-github/pkg/plugin/keyutil.TestReadRSAPrivateKey.func1(0xc0003f8000)
        /usr/local/google/home/qinhang/go/src/github.com/abcxyz/jvs-plugin-github/pkg/plugin/keyutil/pem_test.go:59 +0x9d
testing.tRunner(0xc0003f8000, 0xc000113540)
        /usr/local/google/home/qinhang/Development/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 21
        /usr/local/google/home/qinhang/Development/go/src/testing/testing.go:1648 +0x3ad
FAIL    github.com/abcxyz/jvs-plugin-github/pkg/plugin/keyutil  0.207s

```

Reply:
+1, we should check the gotPK when gotPK == nil and when it != nil.

Reply:
I have updated the code to be:
1. if both gotPK and wantPK are nil, no operation is needed.
2. if they are both not nil, compare they.
3. if only one of them is nil, use `t.Errorf`